### PR TITLE
Implement Draft 2 of GdsSession API

### PR DESCRIPTION
**Pull request by Mats-SX**

### CODE REVIEW

File: graphdatascience/tests/unit/conftest.py [hunk 1]
```diff
@@ -114,9 +113,16 @@ def gds(runner: CollectingQueryRunner) -> Generator[GraphDataScience, None, None
 
 
 @pytest.fixture
-def aura_gds(runner: CollectingQueryRunner) -> Generator[AuraGraphDataScience, None, None]:
+def aura_gds(runner: CollectingQueryRunner, mocker: MockerFixture) -> Generator[AuraGraphDataScience, None, None]:
+    mocker.patch("graphdatascience.query_runner.neo4j_query_runner.Neo4jQueryRunner.create", return_value=runner)
+    mocker.patch(
```
Comment (FlorentinD) [inline graphdatascience/tests/unit/conftest.py:118]: ah thats how you got around it. Could be an idea as well for the other unit tests.
But also increases the test-runtime to patch a lot

**Review by FlorentinD — COMMENTED** [review]
nice improvements :) 

my main comment is around twice calling list_instances now.

File: graphdatascience/gds_session/gds_sessions.py [hunk 2]
```diff
@@ -126,23 +147,29 @@ def _change_initial_pw(self, gds_url: str, gds_user: str, initial_pw: str, new_p
...
+    def _construct_client(
+        self, session_name: str, gds_url: str, db_connection: DbmsConnectionInfo
+    ) -> AuraGraphDataScience:
         return AuraGraphDataScience(
-            endpoint=gds_url, auth=(gds_user, gds_pw), aura_db_connection_info=self._db_credentials
+            gds_session_connection_info=DbmsConnectionInfo(
+                gds_url, GdsSessions.GDS_SESSION_USER, db_connection.password
+            ),
+            aura_db_connection_info=db_connection,
+            delete_fn=lambda: self.delete(session_name),
```
[the remark this answers is not in the document]
Comment (soerenreichardt) [reply graphdatascience/gds_session/gds_sessions.py:158]: We did not decide on that, but this does not auto-close sessions

File: graphdatascience/gds_session/gds_sessions.py [hunk 3]
```diff
@@ -126,23 +147,29 @@ def _change_initial_pw(self, gds_url: str, gds_user: str, initial_pw: str, new_p
...
         )
 
+    @classmethod
+    def session_name(cls, instance_name: str) -> str:
```
[the remark this answers is not in the document]
Comment (soerenreichardt) [reply graphdatascience/gds_session/gds_sessions.py:162]: It is used in the `SessionInfo` dataclass but should not be used by any user. maybe it should go into some helper class then

File: graphdatascience/gds_session/gds_sessions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mats-SX) [reply graphdatascience/gds_session/gds_sessions.py:152]: We want a TTL feature, but we don't want to build it for AuraDS instances. We'll build it into the real GDS Sessions.

File: graphdatascience/gds_session/gds_sessions.py [hunk 4]
```diff
@@ -1,49 +1,66 @@
...
+        db_instance_id = AuraApi.extract_id(self._db_connection.uri)
         db_instance = self._aura_api.list_instance(db_instance_id)
         if not db_instance:
-            raise ValueError(f"Could not find Aura instance with the uri `{self._db_credentials.uri}`")
+            raise ValueError(f"Could not find Aura instance with the uri `{self._db_connection.uri}`")
 
         available_memory_configurations = self._aura_api.list_available_memory_configurations()
+
+        if isinstance(memory, SessionSizeByMemory):
+            memory = str(memory.value)
```
[the remark this answers is not in the document]
Comment (soerenreichardt) [reply graphdatascience/gds_session/gds_sessions.py]: we removed str now

**Pull request merged**